exec/cnex/lib/sqlite.c

```diff
@@ -0,0 +1,404 @@
+#ifdef _WIN32
+#define _CRT_SECURE_NO_WARNINGS
+#endif
+#include "sqlite.h"
+
+#include <inttypes.h>
+
+#include "assert.h"
+#include "cell.h"
+#include "exec.h"
+#include "nstring.h"
+#include "number.h"
+#include "object.h"
+#include "stack.h"
+
+#include "sqlite3.h"
+
+
+
+static Dictionary *Cursors;
+
+void sqlite_initModule(void)
+{
+    Cursors = dictionary_createDictionary();
+}
+
+
+
+inline static Object *check_database(Cell *pdb)
+{
+    if (pdb->type != cObject || pdb->object == NULL || pdb->object->ptr == NULL) {
+        return NULL;
+    }
+    return pdb->object;
+}
+
+
+void object_releaseDatabaseObject(Object *o)
+{
+    if (o != NULL) {
+        assert(o->refcount > 0);
+        o->refcount--;
+        if (o->refcount <= 0) {
+            if (o->ptr != NULL) {
+                int sqliteStatus = sqlite3_close(o->ptr);
+                assert(sqliteStatus == SQLITE_OK);
+                o->ptr = NULL;
+            }
+            free(o);
+        }
+    }
+}
+
+Cell *object_databaseObjectToString(Object *self)
+{
+    char s[32];
+    snprintf(s, 32, "<SQLITE %" PRId64 ">", (int64_t)self->ptr);
```

Comment:
Use `sizeof(s)` here instead of repeating the `32` value

Reply:
Fixed.